python/pyiceberg/io/pyarrow.py

```diff
@@ -317,6 +312,8 @@ def _get_fs(self, scheme: str) -> FileSystem:
             if proxy_uri := self.properties.get(S3_PROXY_URI):
                 client_kwargs["proxy_options"] = proxy_uri
 
+            from pyarrow.fs import S3FileSystem
```

Comment:
I guess we can move it to line 304 (after the if confirms we need to use the S3FileSystem)? feels a bit more readable rather than importing more lazily. but not strongly opinionated

Reply:
Good one, I think I like that more as well 👍🏻